diff --git a/compiler/ghc.mk b/compiler/ghc.mk
index 37a026c..77da9a7 100644
--- a/compiler/ghc.mk
+++ b/compiler/ghc.mk
@@ -554,13 +554,9 @@ compiler_stage2_dll0_MODULES = \
 	VarEnv \
 	VarSet
 
-ifeq "$(GhcWithInterpreter)" "YES"
-# These files are reacheable from DynFlags
-# only by GHCi-enabled code (see #9552)
 compiler_stage2_dll0_MODULES += \
 	ByteCodeTypes \
 	InteractiveEvalTypes
-endif
 
 compiler_stage2_dll0_HS_OBJS = \
     $(patsubst %,compiler/stage2/build/%.$(dyn_osuf),$(subst .,/,$(compiler_stage2_dll0_MODULES)))
